From 9ee3c336014dc3e2c1b3de60acd68dce7f458e3d Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Mon, 29 May 2023 10:24:38 +0800
Subject: [PATCH 213/263] [Huawei] mm/dpool: Bail out on migrate failed

Offering: HULK
hulk inclusion
category: feature
bugzilla: 188703

--------------------------------

Bail out on migrate failed to avoid endless warn message if
do_migrate_range failed.

Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
---
 mm/dynamic_hugetlb.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/mm/dynamic_hugetlb.c b/mm/dynamic_hugetlb.c
index beb501df6610..ac3ad3e79280 100644
--- a/mm/dynamic_hugetlb.c
+++ b/mm/dynamic_hugetlb.c
@@ -1327,12 +1327,16 @@ static int migrate_used_pages(struct dhugetlb_pool *hpool)
 				/* Count used poisoned pages */
 				hpool->nr_poisoned_pages++;
 
-			ret |= do_migrate_range(pfn, pfn+1);
-
+			ret = do_migrate_range(pfn, pfn+1);
 			spin_lock(&hpool->lock);
+
+			/* bail out on migrate failed */
+			if (ret)
+				goto out;
 		}
 	}
 
+out:
 	if (ret)
 		pr_err("dynamic_hugetlb: some 4K pages are still in use, delete memcg: %s failed!\n",
 				hpool->attach_memcg->css.cgroup->kn->name);
-- 
2.17.1

